src/js/models/menubar.js

```diff
@@ -238,6 +238,7 @@ define(function (require, exports, module) {
             templateMenu = fileMenu.submenu[templateIndex],
             templateSubmenu = templateMenu.submenu;
 
+        templateSubmenu.length = 0;
```

Comment:
because we need to make sure we clear the current submenu in this case to avoid duplicate entries.